File: string_to_integer/string_to_integer.go
package stringtointeger

import "strings"

func myAtoi(s string) int {
	s = strings.Trim(s, " ")
	if len(s) == 0 {
		return 0
	}

	var negative bool
	if s[0] == '-' {
		negative = true
	}

	s = strings.Trim(s, "+-")
	if len(s) == 0 {
		return 0
	}

	if negative {
		return -1
	}

	return 1
}

File: java/src/main/java/stringtointeger/StringToInteger.java
package stringtointeger;

public class StringToInteger {
    public static void main(String[] args) {
        var sti = new StringToInteger();

        System.out.println(sti.myAtoi(""));
        System.out.println(sti.myAtoi("   "));
        System.out.println(sti.myAtoi("-"));
    }

    public int myAtoi(String s) {
        char[] str = s.toCharArray();

        for (char c : str) {
            if (c == ' ') {
                continue;
            }
            if (c == '-') {
                return -1;
            }
            if (c == '+') {
                return 0;
            }
        }

        return 0;
    }
}

